src/Me/Modals/MentorshipRequest.js

```diff
@@ -0,0 +1,187 @@
+`;
+
+const FormFields = styled.div`
+  display: flex;
+  flex-wrap: wrap;
+  justify-content: space-between;
+  align-items: stretch;
+  padding: 0 5px;
+  width: 100%;
+`;
+
+const ExtendedFormField = styled(FormField)`
+  flex: 1 1 100%;
+  /* max-width: 355px; */
+  display: flex;
+  flex-direction: column;
+  justify-content: flex-start;
+
+  & label {
+    color: #4f4f4f;
+  }
+
+  width: 100%;
+  min-width: 15rem;
+
+  @media ${desktop} {
+    flex: ${props => (props.customFormField ? '1 1 100%' : '1 1 40%')};
+    max-width: ${props => (props.customFormField ? 'unset' : '355px')};
+  }
+`;
+
+const model = {
+  myBackground: {
+    label: 'My Background',
+    type: 'longtext',
+    defaultValue: '',
+    placeholder: 'Tell the mentor about yourself.',
+  },
+  myExpectations: {
+    label: 'My Expectations',
+    type: 'longtext',
+    defaultValue: '',
+    placeholder: 'What do you expect from this mentorship?',
+  },
+  message: {
+    label: 'Message',
+    type: 'longtext',
+    defaultValue: '',
+    validate: value => !!value,
+    required: true,
+    placeholder: 'Anything else you want to say?',
+  },
+};
+
+const MentorshipRequest = ({ mentor, closeModal }) => {
+  const [confirmed, setConfirmed] = useState(false);
+  const [mentorshipRequestDetails, setMentorshipRequestDetails] = useState({});
+
+  // text fields onChange function
+  const handleInputChange = (fieldName, value) => {
+    setMentorshipRequestDetails({
+      ...mentorshipRequestDetails,
+      [fieldName]: value,
+    });
+  };
+
+  const formField = (fieldName, config) => {
+    switch (config.type) {
+      case 'longtext':
+        return (
+          <ExtendedFormField
+            key={fieldName}
+            label={config.label}
+            helpText={config.helpText}
+          >
+            <Textarea
+              name={fieldName}
+              onChange={e => handleInputChange(fieldName, e.target.value)}
+              required={config.required}
+              placeholder={config.placeholder}
+            />
+          </ExtendedFormField>
+        );
+      default:
+        return <></>;
+    }
+  };
+
+  // validate form details
+  const validate = () => {
+    const errors = [];
+
+    Object.entries(model).forEach(([field, config]) => {
+      if (
+        config.validate &&
+        !config.validate(mentorshipRequestDetails[field])
+      ) {
+        errors.push(config.label);
+      }
+    });
+
+    if (errors.length) {
+      toast.error(
+        `The following fields is missing or invalid: ${errors.join(', ')}`
+      );
+    }
+    return !errors.length;
+  };
+
+  const onSubmit = async e => {
+    e.preventDefault();
+    if (!validate()) {
+      return;
+    }
+    try {
+      const requestResult = await sendMentorshipRequest(
+        mentor.mentor,
+        mentorshipRequestDetails
+      );
+      if (requestResult) {
+        toast.success(messages.CARD_APPLY_REQUEST_SUCCESS);
+        setConfirmed(true);
+      } else {
```

Comment:
We can ignore this because the API is already shows the toast 

https://github.com/Coding-Coach/find-a-mentor/blob/6c8e3da92f6981f59d52f4a5cf64dff0357df2b0/src%2Fapi%2Findex.js#L49-L55